window/src/os/x11/window.rs

```diff
@@ -559,14 +561,46 @@ impl XWindowInner {
         // without the human needing to interact with the window.
         let delay = self.config.focus_change_repaint_delay;
         if delay != 0 {
+
             let window_id = self.window_id;
+
+            let width = self.width;
+            let height = self.height;
+
             promise::spawn::spawn(async move {
+
                 async_io::Timer::after(std::time::Duration::from_millis(delay)).await;
-                XConnection::with_window_inner(window_id, |inner| {
+                XConnection::with_window_inner(window_id, move |inner| {
+
                     inner.sure_about_geometry = false;
                     inner.invalidate();
+
+                    let new_geom = inner.conn().wait_for_reply(inner.conn().send_request(
+                        &xcb::x::GetGeometry {
+                            drawable: xcb::x::Drawable::Window(inner.window_id),
+                        },
+                    ))?;
+
+                    let coords = inner.conn().wait_for_reply(inner.conn().send_request(
+                            &xcb::x::TranslateCoordinates {
+                                src_window: inner.conn().root,
+                                dst_window: window_id,
+                                src_x: 0,
+                                src_y: 0,
+                            }
+                    ))?;
+
+                    if inner.pos_x != coords.dst_x() || inner.pos_y != coords.dst_y() || width != new_geom.width() || height != new_geom.height() {
+                        inner.has_focus = Some(true);
```

Comment:
This may not be accurate; it's assuming that moving the window gains focus.
What I'd suggest instead is to set `inner.verify_focus = true` which will trigger some logic to query the focus state from the WM/X.
That could be done unconditionally as part of this spawn block, and avoid the need to track the window position.
In other words, just unconditionally set `sure_about_geometry = false` and `verify_focus = true`

Reply:
This method was accurate, but you're definitely right that it's better to do it this way. I'd assumed that the inaccurate focus events meant that querying focus directly wouldn't work, which obviously isn't the case.

I've reverted that commit and pushed what you suggested. I've opened another PR with these changes so as not to fill the history with this commit and its reversion.